privacyidea/lib/tokens/hotptoken.py

```diff
@@ -171,10 +202,17 @@ def get_init_detail(self, params=None, user=None):
         # If the init_details contain an OTP key the OTP key
         # should be displayed as an enrollment URL
         otpkey = self.init_details.get('otpkey')
+        # Add two-step initialization parameters to response
+        if is_true(params.get("2stepinit")):
+            response_detail.update(self._get_twostep_parameters())
         if otpkey:
             tok_type = self.type.lower()
             if user is not None:
                 try:
+                    extra_data = {}
+                    # Add two-step initialization parameters to QR codes
+                    if is_true(params.get("2stepinit")):
+                        extra_data.update(self._get_twostep_parameters())
```

Comment:
What is a better pythonic way? I think in this case ``update`` is the same like

      extra_data = self._get_twostep_parameters()

Reply:
815577b943a9090ec4211bfac7bf8655130c3a31 improves the structure a bit, I hope.